docs/addons.md

```diff
@@ -127,12 +127,15 @@ spec:
 
 {{ kops_feature_table(kops_added_default='1.19') }}
 
-Node Termination Handler ensures that the Kubernetes control plane responds appropriately to events that can cause your EC2 instance to become unavailable, such as EC2 maintenance events, EC2 Spot interruptions, ASG Scale-In, ASG AZ Rebalance, and EC2 Instance Termination via the API or Console. If not handled, your application code may not stop gracefully, take longer to recover full availability, or accidentally schedule work to nodes that are going down.
+[Node Termination Handler](https://github.com/aws/aws-node-termination-handler) ensures that the Kubernetes control plane responds appropriately to events that can cause your EC2 instance to become unavailable, such as EC2 maintenance events, EC2 Spot interruptions, and EC2 instance rebalance recommendations. If not handled, your application code may not stop gracefully, take longer to recover full availability, or accidentally schedule work to nodes that are going down.
+
+If `enableSqsTerminationDraining` is enabled Node Termination Handler will operate in Queue Processor mode. In addition to the events mentioned above, Queue Processor mode allows Node Termination Handler to take care of ASG Scale-In, AZ-Rebalance, Unhealthy Instances, EC2 Instance Termination via the API or Console, and more. kOps will provision the necessary infrastructure: an SQS queue, EventBridge rules, and ASG Lifecycle hooks.
```

Comment:
I didn't test this myself, but I believe that enabling `enableSqsTerminationDraining` on an existing cluster that previously had NTH itself enabled, the regular NTH daemonset would not be deleted when replaced by the SQS Deployment. The reverse would also be true.

It might be worth mentioning that the cluster admin will need to manually delete the old resource(s) when toggling this field.

Reply:
We need to clearly document this. 
Luckily I don't think it is a huge problem for both to co-exist, but still ..

Reply:
Deleting the daemonset/deployment is easy enough, but should we try and automate deleting the aws resources when switching from queue processor mode to IMDS mode? Manually deleting the aws resources required for queue processor mode is a subpar experience, but I'm not quite sure how this would fit into the kOps update process

Reply:
It is a feature we need to look into. But right now, kOps does a poor job of tracking addon resources, so it is less than trivial. Addon operators is believed to be the solution to this, but is out of the scope of this PR for sure.

Right now we just have to live with subpar experience.